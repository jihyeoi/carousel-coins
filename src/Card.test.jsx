import { it, expect } from "vitest";

import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";
import image1 from "./image1.jpg";
import image2 from "./image2.jpg";
import image3 from "./image3.jpg";


// smoke test
it("renders without crashing", function () {
    render(<Card
        caption="Photo by Richard Pasquarella on Unsplash"
        src={image1}
        currNum={1}
        totalNum={3} />)
});

// snapshot test
it("matches snapshot", function () {
    const { container } = render(<Card
            caption="Photo by Richard Pasquarella on Unsplash"
            src={image1}
            currNum={1}
            totalNum={3} />);
    expect(container).toMatchSnapshot();
});