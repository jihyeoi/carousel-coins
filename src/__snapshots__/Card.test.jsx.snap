// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="/src/image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      3
      .
    </small>
  </div>
</div>
`;
